service/datastore/internal/src/main/java/org/eclipse/kapua/service/datastore/internal/mediator/DatastoreUtils.java

```diff
@@ -137,6 +137,9 @@ public String getDataIndexName(KapuaId scopeId) {
             sb.append(prefix).append("-");
         }
         String indexName = normalizedIndexName(scopeId.toStringId());
+        if (scopeId.equals(KapuaId.ANY)) {
+            indexName = "*";
```

Comment:
Overriding a previously set value is conceptually wrong - both because you waste the first calculation (however trivial it might be) and also because the code becomes harder to read (If I want to read through in order to figure out the final value of the variable, I have to maintain the cognitive load of the first calculation, only to then discard the process when it gets overwritten).
TL;DR: use an if-then-else construct here.

Also, when comparing unknown variables with known quantities, try to always use the known value as the base for the comparison:
KapuaId.ANY.equals(scopeId) will never throw NPE, and will still work against nulls.

Reply:
I've done this because I assumed that in any case first of all we want to normalize the index to validate its format (even if it's null, like it is done now) , but ok I can do as you suggest